ps2-decompile/callbacks.c

```diff
@@ -1,194 +1,220 @@
-#include <usbd.h>
+#include "callbacks.h"
 
-#include "globals.h"
-#include "shared.h"
+static char report[0x400];
+static const char string_desc[] = {0};
+static const int off_2C30 = 0x2C30;
+static char kb[0x10];
 
 int data_transfer_done(int result, int count, shared_t *g) {
-    int usbmouse_count; // $a2
-    int v4;             // $gp
-    int i;              // [sp+14h] [+14h]
+    int i, offset;
+    (void) count;
 
+#ifdef DEBUG
     if (result)
-        printf("usbmouse%d: %s -> 0x%x\n", g->port1, "sceUsbdInterruptTransfer", result);
+        (void) printf("usbmouse%d: %s -> 0x%x\n", g->port1, "sceUsbdInterruptTransfer", result);
+#endif
+
     if (!result && g->descriptorChecksPassed) {
-        usbmouse_count = g->count + 1;
-        g->count = usbmouse_count;
-        printf("usbmouse%d: count=%d data=(", g->port1, usbmouse_count);
+        g->count += 1;
+
+#ifdef DEBUG
+        (void) printf("usbmouse%d: count=%d data=(", g->port1, g->count);
+#endif
+
         for (i = 0; i < 8; ++i) {
-            // v4 - 32752 = 2CCC
-            printf((const char *)(v4 + 0xFFFF8010), g->unk8[i + 0x7804]);
+            offset = i + 0x7804;
+
+#ifdef DEBUG
+           (void) printf("data at offset %i byte=%c", offset, g->data[offset]);
+#endif
+
             if (g->port) {
-                *(char *)(8 * (g->port - 1) + i + 0x2D20) = g->unk8[i + 0x7804];
+                // TODO figure out what the driver is doing here.
+                // it appears to be a memcpy.
```

Comment:
I do not think this is an accurate comment so please remove it.

What is going on below is accessing an array and assigning a value at some index. It also could be accessing a struct offset.